doc/py_tutorials/py_setup/py_setup_in_ubuntu/py_setup_in_ubuntu.markdown

```diff
@@ -0,0 +1,242 @@
+below. You can either leave it or install it, your call :)
+
+OpenCV comes with supporting files for image formats like PNG, JPEG, JPEG2000, TIFF, WebP etc. But
+it may be a little old. If you want to get latest libraries, you can install development files for
+these formats.
+<br>
+`sudo apt-get install libpng-devel`<br>
+`sudo apt-get install libjpeg-turbo-devel`<br>
+`sudo apt-get install jasper-devel`<br>
+`sudo apt-get install openexr-devel`<br>
+`sudo apt-get install libtiff-devel`<br>
+`sudo apt-get install libwebp-devel`<br>
+
+Several OpenCV functions are parallelized with **Intel's Threading Building Blocks** (TBB). But if
+you want to enable it, you need to install TBB first. ( Also while configuring installation with
+CMake, don't forget to pass -D WITH_TBB=ON. More details below.)
+<br>
+` sudo apt-get install tbb-devel`<br>
+### Downloading OpenCV
+
+Next we have to download OpenCV. You can download the latest release of OpenCV from [sourceforge
+site](http://sourceforge.net/projects/opencvlibrary/). Then extract the folder.
+
+Or you can download latest source from OpenCV's github repo. (If you want to contribute to OpenCV,
+choose this. It always keeps your OpenCV up-to-date). For that, you need to install **Git** first.
+<br>
+`sudo apt-get install git`<br>
+`git clone https://github.com/opencv/opencv.git`
+<br>
+It will create a folder OpenCV in home directory (or the directory you specify). The cloning may
+take some time depending upon your internet connection.
+
+Now open a terminal window and navigate to the downloaded OpenCV folder. Create a new build folder
+and navigate to it.
+<br>
+`mkdir build`<br>
+`cd build`
+<br>
+### Configuring and Installing
+
+Now we have installed all the required dependencies, let's install OpenCV. Installation has to be
+configured with CMake. It specifies which modules are to be installed, installation path, which
+additional libraries to be used, whether documentation and examples to be compiled etc. Below
+command is normally used for configuration (executed from build folder).
+<br>
+`cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local ..`
+<br>
+It specifies that build type is "Release Mode" and installation path is /usr/local. Observe the -D
+before each option and .. at the end. In short, this is the format:
+<br>
+cmake [-D <flag>] [-D <flag>] ..
+<br>
+You can specify as many flags you want, but each flag should be preceded by -D.<br>
+
+So in this tutorial, we are installing OpenCV with TBB. We also build the
+documentation, but we exclude Performance tests and building samples. We also disable GPU related
+modules (since we use OpenCV-Python, we don't need GPU related modules. It saves us some time).<br>
+
+*(All the below commands can be done in a single cmake statement, but it is split here for better
+understanding.)*
+
+-   Enable TBB and Eigen support:
+    <br>
+    `cmake -D WITH_TBB=ON -D WITH_EIGEN=ON ..`
```

Comment:
> WITH_TBB=ON

Not required.

> WITH_EIGEN

Enabled by default, so not required too.

We prefer to suggest OpenCV's configuration defaults at first without long list of "magic" options.